### ISSUE TITLE
environment-parity-standards: Angular/Nx build configurations are not recognised as environment config

### ISSUE THREAD

**Issue by ruleofcodedev** [Rule-Of-Code/RoC#94]
**Version:** 7.20.0
**Law:** `environment-parity-standards` (`BuildEnvironmentAnalyzer`)

## What happens

```
No proper environment detection found
```

The advice attached to it is `Implement NODE_ENV or similar environment
detection` and `Use environment-specific config files`.

An Angular/Nx workspace expresses exactly that, and does it in the build system
rather than in the process environment. Ours declares it twice:

```jsonc
// apps/web/project.json
"build":  { "configurations": { "production": { … }, "development": { … } },
            "defaultConfiguration": "production" },
"serve":  { "configurations": { "production": { … }, "development": { … } } }
```

Those configurations are what select the production budgets, output hashing and
optimisation here. They are the environment-specific configuration the law is
asking for; the checker looks only for `NODE_ENV`, `.env` files, Dockerfiles and
docker-compose, none of which an Angular application uses.

## Why this reads as a false negative rather than a missing feature

The law's own remediation — "use environment-specific config files" — is already
satisfied, so following the advice produces no change and the finding does not
move. That is the same shape as the `chore/` branch fix in 7.20.0: advice that
cannot resolve the finding it is attached to.

An Angular project that adds `NODE_ENV` to satisfy this would be adding a
variable nothing reads, which is ceremony rather than parity.

## Scope, honestly

This site is prerendered and deployed to one target, so its parity risk is small
either way. The report matters because our policy is that a law is `error` or it
is declared not-applicable with a written reason — and neither is true here. The
concern exists, it is met, and the checker cannot see it, so the law sits at
`warning` with a note, which is exactly the state we are trying not to have.

## Suggested direction

Recognise `project.json` / `angular.json` build `configurations` (and the
`defaultConfiguration`) as environment configuration, alongside the existing
`NODE_ENV`, dotenv and Docker signals. `nx.json` presence is a reliable marker
that this is where to look.

If a workspace declares only one configuration, the current finding is fair —
that genuinely is a single-environment build.

**Comment by ruleofcodedev** [comment]
**Fixed — in #97, awaiting CI and merge to `develop`. Ships in the next release.**

Build `configurations` are recognised as environment configuration now, from both `project.json` and a root `angular.json`, and on any target rather than just `build`.

Your boundary is respected exactly as you drew it: **a single configuration is still reported.** That genuinely is a single-environment build, and the finding there is fair.

Your framing decided the fix. The law's own remediation — "use environment-specific config files" — was already satisfied, so the finding could not be moved by doing what it asked. That is the same defect class as the `chore/` branch advice in 7.20.0, and you named the parallel yourself.

One implementation note: this needed the build config **parsed**, not matched as text. "How many named configurations does a target declare" is a question about structure, and the existing shared reader only offers concatenated content. A file that will not parse is skipped rather than guessed at.

On your scope note — thank you for writing it. A report that says "our parity risk here is small either way, this matters because of the warning-limbo policy" is easier to rank honestly than one that overstates the impact.

**Comment by ruleofcodedev** [comment]
Merged to `develop` in #97. Ships in the next release.
